plugin/code_actions.py

```diff
@@ -130,6 +130,8 @@ def request_factory(sb: SessionBufferProtocol) -> Request | None:
                 if diag_sb == sb:
                     diagnostics = diags
                     break
+            # Pull for diagnostics to ensure that server computes them before receiving code action request.
+            sb.do_document_diagnostic_async(view)
```

Comment:
I probably would have put this part under `if request:` in `_collect_code_actions_async` (line 165), because it's not strictly necessary for creating the Request object.

Reply:
makes sense

Ideally perhaps this should be trigger before any code action request. While normal usage should not require that, if someone uses a macro or some sort of automation then it could become a problem then too.

Reply:
Actually this code is used for any code action request so all good.